docs/raw/9-blockchain-interactions.md

```diff
@@ -0,0 +1,186 @@
+---
+permalink: /spec/9
+parent: Raw specs
+title: 9/BLOCKCHAIN-USAGE
+---
+
+# 3/BLOCKCHAIN-USAGE
+
+> Version: 0.1
+>
+> Status: Raw
+>
+> Authors: Andrea Maria Piana <andreap@status.im>
+
+# Status interactions with the blockchain
+
+In this document is document all the interactions that the Status client has
+with the blockchain.
+
+1. [Wallet](#Wallet)
+2. [ENS](#ENS)
+
+## Wallet
+
+The wallet in Status has two main components:
+
+1) Sending transactions
+2) Fetching balance
+
+In the section below are described the `RPC` calls made the nodes, with a brief
+description of their functionality and how it is used by Status.
+
+1. [Sending transactions](#Sending-transactions)
+  - [EstimateGas](#EstimateGas)
+  - [PendingNonceAt](#PendingNonceAt)
+  - [SuggestGasPrice](#SuggestGasPrice)
+  - [SendTransaction](#SendTransaction)
+2. [Fetching balance](#Fetching-balance)
+  - [BlockByHash](#BlockByHash)
+  - [BlockByNumber](#BlockByNumber)
+  - [FilterLogs](#FilterLogs)
+  - [HeaderByNumber](#HeaderByNumber)
+  - [NonceAt](#NonceAt)
+  - [TransactionByHash](#TransactionByHash)
+  - [TransactionReceipt](#TransactionReceipt)
+
+### Sending transactions
+
+#### EstimateGas
+
+EstimateGas tries to estimate the gas needed to execute a specific transaction based on
+the current pending state of the backend blockchain. There is no guarantee that this is
+the true gas limit requirement as other transactions may be added or removed by miners,
+but it should provide a basis for setting a reasonable default.
+
+```
+func (ec *Client) EstimateGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L499
+
+
+#### PendingNonceAt
+`PendingNonceAt` returns the account nonce of the given account in the pending state.
+ This is the nonce that should be used for the next transaction.
+
+ ```
+func (ec *Client) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L440
+
+
+#### SuggestGasPrice
+
+`SuggestGasPrice` retrieves the currently suggested gas price to allow a timely
+execution of a transaction.
+
+```
+func (ec *Client) SuggestGasPrice(ctx context.Context) (*big.Int, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L487
+
+#### SendTransaction
+
+`SendTransaction` injects a signed transaction into the pending pool for execution.
+
+If the transaction was a contract creation use the TransactionReceipt method to get the
+contract address after the transaction has been mined.
+
+
+```
+func (ec *Client) SendTransaction(ctx context.Context, tx *types.Transaction) error 
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L512
+
+### Fetching balance
```

Comment:
Intro para, be explicit about goals here with ERC20 and other relevant standards. Also that we SHOULD (?) care about historical data, etc

E.g. what do we do for something like collectibles? Is that covered in this spec?